### PR TITLE
[13.0][IMP][WIP] stock_barcodes: Improvements and restyling

### PR DESCRIPTION
**Pull request by sergio-teruel**
https://github.com/OCA/stock-logistics-barcode/pull/347 superseded

ping @etobella @JordiBForgeFlow @fshah-initos @olgamarcocb
This PR is in wip.. I hope that the next week we work hard to finish...

Improvements:
* Specific customizable menu visible in wizard interface
image
* Configurable options by type of operation
image
* New guided mode
image
* Restyling to adapt to mobile barcode readers

Other related modules:
* https://github.com/OCA/stock-logistics-barcode/pull/349
* https://github.com/OCA/stock-logistics-barcode/pull/327
* https://github.com/OCA/stock-logistics-barcode/pull/329
* https://github.com/OCA/stock-logistics-barcode/pull/350

### CODE REVIEW

**Comment by sergio-teruel** [comment]
The PR https://github.com/OCA/stock-logistics-barcode/pull/348 is ready for review

**Comment by etobella** [comment]
the old behavior is not working properly. On runbot, I configured a product package, created a Picking, started scanning on the picking but it did nothing:

image

Also, navigation is deleted, making it hard to work on this case....

**Comment by MiquelRForgeFlow** [comment]
please rebase

**Comment by pedrobaeza** [comment]
Done, the translabot made me a little headache for solving conflicts.

**Comment by etobella** [comment]
Navigation is still lost. I understand that in some customers it can be neccessary to remove it, but not in all. So please, return the navigation. Also, deleting the navigation bar creates som issues.
Peek 11-11-2021 09-22
Also, If I take a picking It is required to scan the Location, but it could take it automatically from the Lot, isn't it?

**Comment by pedrobaeza** [comment]
For the navigation thing, did you figure out how to solve it in your PR? Can you point to that?

**Comment by pedrobaeza** [comment]
My colleague is telling me that the navigation bar is removed on purpose for gaining such height on small screen devices like mobiles, so if you see it useful, you have to add it as a configurable option.

**Comment by sergio-teruel** [comment]
We are working on remove panel navigation using settings

**Comment by etobella** [comment]
My point is, if you access the view you loose the navigation bar, that can be fine. But if you return to any other view, the navigation bar is not shown anymore.

**Comment by manuelcalerosolis** [comment]
My test.
Create a product
image

Create a purchase
image

Go to Inventory
image

Go to the transfer and press "Scan Barcode"
image

image

Scan the barcode
image

And nothing happens

**Comment by sergio-teruel** [comment]
Hi @manuelcalerosolis You must setting up the barcode options here "Inventory > Configuration > Barcode options".
And link a option to a picking type 
We still no documentation.

**Comment by LoisRForgeFlow** [comment]
@sergio-teruel Anything else missing before merging this one?

**Comment by sergio-teruel** [comment]
@LoisRForgeFlow I want change the tests... during this week i will do it!

**Comment by etobella** [comment]
@sergio-teruel Any news when you will finish this? It is a blocking PR. Maybe we could propose a merge and then apply improvements on separate PRs?

**Comment by sergio-teruel** [comment]
Hi... @etobella I think that we can merge...
I already finished the tests...

**Review by LoisRForgeFlow — APPROVED** [review]
Tested and LGTM in general

If you squash your commits, I think we can move forward :rocket:

File: stock_barcodes/wizard/stock_barcodes_read_picking_views.xml [hunk 1]
```diff
@@ -63,78 +69,164 @@
...
+                        name="pending_move_ids"
+                        options="{'no_open': True, 'always_reload': True}"
+                        nolabel="1"
+                        force_save="1"
+                        mode="tree,kanban"
+                    >
+                        <tree string="Pending moves">
+                            <field name="state" />
+                            <field name="product_id" />
+                            <field name="product_uom_qty" />
```
Comment (ChrisOForgeFlow) [inline stock_barcodes/wizard/stock_barcodes_read_picking_views.xml:147]:
```suggestion
                            <field name="product_uom_qty" />
                            <field name="line_ids" invisible="1"/>
```

File: stock_barcodes/wizard/stock_barcodes_read_todo_view.xml [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+                <field name="location_dest_id" />
+                <field name="location_dest_name" />
+                <field name="product_id" />
+                <field name="lot_id" />
+                <field name="uom_id" />
+                <field name="package_id" />
+                <field name="result_package_id" />
+                <field name="package_product_qty" />
+                <field name="product_uom_qty" />
+                <field name="qty_done" />
```
Comment (ChrisOForgeFlow) [inline stock_barcodes/wizard/stock_barcodes_read_todo_view.xml:20]:
```suggestion
                <field name="qty_done" />
                <field name="line_ids" invisible="1"/>
```

**Review by ChrisOForgeFlow — CHANGES_REQUESTED** [review]
Please add line_ids on kanban and tree view of model 'wiz.stock.barcodes.read.todo' to trigger onchange on qty_done, field should be on view to ensure compute trigger on customer side

**Comment by LoisRForgeFlow** [comment]
@sergio-teruel Could you squash your commits?

**Comment by sergio-teruel** [comment]
@LoisRForgeFlow Done!!

**Comment by LoisRForgeFlow** [comment]
Thanks!

@etobella good to go? :)

**Review by etobella — APPROVED** [review]
Let's go for live!!

**Comment by sergio-teruel** [comment]
:bomb: jajaj

**Comment by MiquelRForgeFlow** [comment]
But @ChrisOForgeFlow suggestions are not applied yet :(

**Comment by sergio-teruel** [comment]
Changes suggested by @ChrisOForgeFlow applied!

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by LoisRForgeFlow** [comment]
Let's go! :see_no_evil: 

/ocabot merge major

**Pull request merged**